### ISSUE TITLE
211668809 BA with only Birth date shows error when uploaded against an embargo (Eric D)

### ISSUE THREAD

**Issue by ghost** [FreeUKGen/FreeUKRegProductIssues#1447]
Issue reported by **REGManager** at 2021-09-15 20:53:29 UTC
Time: 2021-09-15T20:48:36+00:00
Session ID: 78e8b6bf77e3c057741f546987406563
Problem Page URL: [/freereg1_csv_entries?locale=en](/freereg1_csv_entries?locale=en)
Previous Page URL: [https://www.freereg.org.uk/freereg1_csv_files/60baf089f493fd98fb456805?locale=en](https://www.freereg.org.uk/freereg1_csv_files/60baf089f493fd98fb456805?locale=en)
Reported Issue:
In the attached the last four entries of the file show a new error of embargo year cannot be computed.  This is because there is no BA date.
https://www.freereg.org.uk/uploads/feedback/screenshot/61425d49f493fd3aa6d0eb81/Capture.JPG    

![Screenshot](https://www.freereg.org.uk)

**Comment by edickens** [comment]
Still needs fixing, but is a rare occurance.  It does mean that some entries may not be shown when they should, but they could also be embargoed entries, however when the embargo ends they will not be shown.

**Comment by Gev1990123** [comment]
Opened pull request https://github.com/FreeUKGen/MyopicVicar/pull/2822

Currently undergoing testing.

**Comment by Vino-S** [comment]
Instead of returning false, Embargo should check other dates in the Record if it falls in embargo year range. For example: comfirmation_date, burial_date, birth_date, death_date, baptism_date

**Comment by Gev1990123** [comment]
Have updated to accomodate the above. Pull Request has been updated, pending testing.

**Comment by Vino-S** [comment]
@stsccfr ready for testing on test2

**Comment by stsccfr** [comment]
The embargo I created on test2 doesn't seem to be applied.  I uploaded a new file called STSCMBBU.CSV (attached here) 

[STSCMBBU.CSV](https://github.com/user-attachments/files/23060124/STSCMBBU.CSV)

which contains 5 burial records dating from 1906 to 1958.  I created a 100-year embargo, which should have embargoed 2 records from this file, which have dates of 1951 and 1958.  But no records are showing up as being embargoed.

**Comment by Gev1990123** [comment]
@stsccfr i have tried to reproduce this on my development env, and i get the two records which are embargoed: 

<img width="3468" height="1389" alt="Image" src="https://github.com/user-attachments/assets/5c18f28a-9f2d-4b01-82be-6aeb549f8d7c" />

I have also tried to reproduce on test2 (dev.freereg.org.uk) and again i see the two records which are embargoed:

<img width="3678" height="1266" alt="Image" src="https://github.com/user-attachments/assets/24397e75-6905-486e-808b-0f796a47050d" />

**Comment by stsccfr** [comment]
I still see no embargoed entries for my STSCMBBU file on dev (test2), but I
*do* see them for your own STSCMBBU file!  When you tested this, did you
create any embargo rules, or did you just upload a copy of my file into
your own account?  If the latter, then it seems we are only applying
embargo rules to new uploads, not to any existing files.

In the document written by Kirk Dawson who I think implemented the embargo
system, it says:

Once created all records for batches in the register will be reprocessed in
the background overnight; during testing on test3 it will run every 30
minutes on the hr and half hour. You will receive an email with a detailed
report once completed.

So it seems that the embargo is intended to apply to all existing records,
which makes sense and I'm sure is what we want, but that doesn't seem to be
happening any more.

**Comment by Gev1990123** [comment]
I just uploaded the file again under my account on dev (test2) and no additional embargo rules where created. 

When the embargo rule was created, was this created before or after uploading the file?

**Comment by stsccfr** [comment]
Before uploading, did you change at least one of the record dates so that
it would be within the period of the embargo?

I can't remember the order of events with regard to my STSCMBBU file, but I
just now uploaded a second one, STSCMBBU1 with some new records, and I *do*
see that two of them are under embargo.  So there's still reason to believe
that we aren't processing existing records, only new ones.

**Comment by Gev1990123** [comment]
No, i dont recall changing any of the records. 

So to confirm, if we was to add an embargo rule to a register it should re-process all the existing records (and any new records added) against that embargo rule?

**Comment by stsccfr** [comment]
Yes, a rule should apply to all existing and all new records; otherwise the
embargo will be only partial.  But I don't know how the reprocessing of
existing records is supposed to happen.  It appears from Kirk's document
that it might be via a rake task that runs only periodically so as not to
overload the server, but I don't know whether that is still a concern.

**Comment by Gev1990123** [comment]
I can take a look over the coming days, but do rake tasks run on the dev and test servers?

**Comment by stsccfr** [comment]
I didn't even think of that, but I'll bet they aren't run on dev/test
automatically as part of our usual setup.

**Comment by Gev1990123** [comment]
I beleive the file is reprocess_batches_for_a_county.rake which reproccess embargo rules via the method update_create_search_record within search_record.rb

I do not have access to the backend of the dev server, so i am unable to confirm if this rake task is run on a schedule or not.

**Comment by DeniseColbert** [comment]
@Vino-S to answer when back in office

**Comment by Vino-S** [comment]
Ready for testing on DEV

**Comment by stsccfr** [comment]
@Vino-S Still not working on DEV.  There are two CSV files for the Parish Register for the church of St Mary in Calton, STS and both have the name STSCMBBU.CSV. One is in my account and the other in Matt's.  These two files contain the same records, and two of the records in these files should be under embargo, but only the two records in Matt's file are showing up as embargoed.  When an embargo is created, that should trigger a rake task to reprocess all of the records in all CSV files for that register.  Currently it seems that the task is only processing new uploads, not existing files already on the server.

**Comment by stsccfr** [comment]
The rake task to impose embargoes is set by a cron job to run daily every few hours.  Now that the rake task has been run on DEV, the records for the parish register of Calton St Mary have all been processed and the embargoed ones are showing up correctly.  The code changes can be deployed to Live.
